templates/ansible/facts.erb

```diff
@@ -87,6 +89,9 @@ def main():
   })), 12)
 -%>
     )
+<% if version_added(object, :facts) < '2.9' && module_suffix == 'facts' -%>
+    module.deprecate("The '<%= module_name(object) -%>_facts' has been renamed to '<%= module_name(object) -%>_info'", version='2.13')
```

Comment:
This should only be called if the module is actually used with the old name.

Reply:
Ah wait. You now have to copies of the same module around? It would be better if there's only one source file (_info), and the other is a symlink. Then it is clear that this is a rename.

Reply:
Or if you want to deprecate the old name, you should rename the old file to `_oldname_facts.py` (leading underscore) and add deprecation info (see for example https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/f5/_bigip_facts.py#L33).

Reply:
Ah, that would be a much better idea. So, just to be clear:

* Module lives at "gcp_whatever_info.py"
* The old version would be a symlink to above with name "_gcp_whatever_facts.py".
* Some logic in "gcp_whatever_info.py" that would call `module.deprecate` if the filename is "_gcp_whatever_facts.py"

Reply:
The old version would be a symlink with name "_gcp_whatever_facts.py"; the underscore tells Ansible it is deprecated. (And the symlink will be removed four versions later, i.e. for Ansible 2.13.)